src/Migrator/MigrationWorkflow.cs

```diff
@@ -0,0 +1,111 @@
+    /// used outside of internal calls.
+    /// </para>
+    /// <param name="runType">The run type.</param>
+    /// <param name="objects">The objects.</param>
+    /// <returns>The <see cref="Task"/>.</returns>
+    [WorkflowRun]
+    public async Task RunAsync(RunType runType = RunType.Bootstrap, string[]? objects = null)
+    {
+        switch (runType)
+        {
+            case RunType.Bootstrap:
+                await RunEntryAsync(objects ?? [], Workflow.CancellationToken);
+                break;
+            case RunType.Migration:
+                ArgumentNullException.ThrowIfNull(objects);
+                await RunMigrationAsync(objects[0], Workflow.CancellationToken);
+                break;
+            default:
+                throw new ArgumentOutOfRangeException(nameof(runType), runType, "Invalid MigrationRunner option");
+        }
+    }
+
+    private async Task RunEntryAsync(string[] assemblies, CancellationToken cancellationToken)
+    {
+        cancellationToken.ThrowIfCancellationRequested();
+        var workflowId = Workflow.Info.WorkflowId;
+
+        var types = await Workflow.ExecuteLocalActivityAsync<MigrationActivities, IEnumerable<Type>>(
+            act => act.GetTypesAsync(assemblies),
+            new LocalActivityOptions
+            {
+                ActivityId = $"{workflowId}_{nameof(MigrationActivities)}",
+                ScheduleToCloseTimeout = TimeSpan.FromMinutes(5),
+                CancellationToken = cancellationToken,
+            });
+
+        foreach (var migrationType in types)
+        {
+            var nextType = migrationType.ToString();
+
+            await Workflow.ExecuteChildWorkflowAsync<MigrationWorkflow>(
+                s => s.RunAsync(
+                    RunType.Migration,
+                    new[] { nextType }),
+                new ChildWorkflowOptions
+                {
+                    Id = $"{workflowId}_{migrationType.Name}",
+                    CancellationToken = cancellationToken,
+                });
+        }
+    }
+
+    /// <summary>
+    /// Run Migration.
+    /// </summary>
+    /// <param name="type">The type to run.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>The <see cref="Task"/>.</returns>
+    private async Task RunMigrationAsync(string type, CancellationToken cancellationToken)
+    {
+        cancellationToken.ThrowIfCancellationRequested();
+
+        var potentialMatches = GlobalReflector.GetMigrations(Array.Empty<Assembly>());
+
+        if (potentialMatches.Count() != 0)
+        {
+            var first = potentialMatches.First(w => w.ToString() == type);
+
+            if (Activator.CreateInstance(first) is not IMigration migration)
+            {
+                throw new InvalidOperationException("Type is not a migration object");
+            }
+
+            await migration.ExecuteAsync(cancellationToken);
```

Comment:
If this is running the workflow for the migration, it's going to run in the context of the parent migration.

I think it should be a child workflow.

Look at dynamic workflows potentially for this or have a workflow that takes in the assembly and migration name to execute and have that execute it so it shows as a child workflow and has traceability.